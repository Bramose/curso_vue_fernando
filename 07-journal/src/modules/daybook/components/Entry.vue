<template>
  <div @click="$router.push({ name: 'entry', params: { id: 10 } })" class="entry-container mb-3 pointer p-2">
      <div class="entry-title d-flex">
          <span class="text-success fs-5 fw-bold">08</span>
          <span class="mx-1 fs-5">Noviembre</span>
          <span class="mx-2 fw-light">2021, lunes</span>
      </div>
      <div class="entry-description">
        Culpa excepteur labore mollit veniam occaecat magna laboris magna veniam occaecat sit. Eu voluptate non elit laborum officia do fugiat. Do id Lorem ex occaecat esse elit.Elit ea veniam pariatur culpa culpa ad. Magna quis nulla esse culpa elit exercitation nulla occaecat et incididunt incididunt. Non tempor ex ullamco dolor elit.
      </div>
  </div>
</template>

<style lang="scss" scoped>
.entry-container{
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;
    &:hover{
        background-color: lighten($color:grey, $amount: 45);
        transition: 0.2s all ease-in;
    }
    .entry-description{
        font-size: 12px;
    }
}
</style>