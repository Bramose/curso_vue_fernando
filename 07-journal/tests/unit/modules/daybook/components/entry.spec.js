import { shallowMount } from '@vue/test-utils'
import Entry from '@/modules/daybook/components/Entry.vue'
import { journalState } from '../../../mock-data/test-journal-state.js'


describe('pruebas con el entry component', () => {
    const mockRouter = {
        push: jest.fn()
    }

    const entry = journalState.entries[0]

    const wrapper = shallowMount(Entry, {
        props: {
            entry
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })

    test('debe de hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe de redireccionar al hacer click en el entry-container', () => {
        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')
        expect(mockRouter.push).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: entry.id
            }
        })
    })

    test('pruebas en las propiedades computadas', () => {
        const { day, month, yearDay } = wrapper.vm
        expect(day).toBe(15)
        expect(month).toBe('Noviembre')
        expect(yearDay).toBe('2021, Lunes')
    })
})