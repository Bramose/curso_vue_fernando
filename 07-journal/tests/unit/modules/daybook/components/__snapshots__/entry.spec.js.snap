// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con el entry component debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">15</span><span class="mx-1 fs-5">Noviembre</span><span class="mx-2 fw-light">2021, Lunes</span></div>
  <div class="entry-description">Que voy hacer hoy
  </div>
</div>
`;
